iothub_client/src/iothubtransporthttp.c

```diff
@@ -1540,199 +1744,76 @@ static void DoEvent(HTTPTRANSPORT_HANDLE_DATA* handleData, HTTPTRANSPORT_PERDEVI
-                                {
-                                    if (HTTPHeaders_ReplaceHeaderNameValuePair(clonedEventHTTPrequestHeaders, IOTHUB_CORRELATION_ID, corrId) != HTTP_HEADERS_OK)
-                                    {
-                                        LogError("unable to HTTPHeaders_ReplaceHeaderNameValuePair");
-                                        goOn = false;
-                                    }
-                                }
-
-                                // Codes_SRS_TRANSPORTMULTITHTTP_09_001: [ If the IoTHubMessage being sent contains property `content-type` it shall be added to the HTTP headers as "iothub-contenttype":"value". ]
-                                userDefinedContentType = IoTHubMessage_GetContentTypeSystemProperty(message->messageHandle);
-                                if (goOn && userDefinedContentType != NULL)
-                                {
-                                    if (HTTPHeaders_ReplaceHeaderNameValuePair(clonedEventHTTPrequestHeaders, IOTHUB_CONTENT_TYPE_D2C, userDefinedContentType) != HTTP_HEADERS_OK)
-                                    {
-                                        LogError("unable to HTTPHeaders_ReplaceHeaderNameValuePair (content-type)");
-                                        goOn = false;
-                                    }
-                                }
-
-                                // Codes_SRS_TRANSPORTMULTITHTTP_09_002: [ If the IoTHubMessage being sent contains property `content-encoding` it shall be added to the HTTP headers as "iothub-contentencoding":"value". ]
-                                contentEncoding = IoTHubMessage_GetContentEncodingSystemProperty(message->messageHandle);
-                                if (goOn && contentEncoding != NULL)
-                                {
-                                    if (HTTPHeaders_ReplaceHeaderNameValuePair(clonedEventHTTPrequestHeaders, IOTHUB_CONTENT_ENCODING_D2C, contentEncoding) != HTTP_HEADERS_OK)
-                                    {
-                                        LogError("unable to HTTPHeaders_ReplaceHeaderNameValuePair (content-encoding)");
-                                        goOn = false;
-                                    }
-                                }
-
-                                if (!goOn)
+                                if (set_system_properties(message, clonedEventHTTPrequestHeaders) != 0)
                                 {
                                     /*Codes_SRS_TRANSPORTMULTITHTTP_17_079: [If any HTTP header operation fails, _DoWork shall advance to the next action.] */
                                 }
                                 else
                                 {
-                                    BUFFER_HANDLE toBeSend = BUFFER_new();
+                                    BUFFER_HANDLE toBeSend = BUFFER_create(messageContent, originalMessageSize);
```

Comment:
>originalMessageSize [](start = 91, length = 19)

Not necessairly code you've touch, but I'm having trouble getting head around the MAXIMUM_PAYLOAD_OVERHEAD business.  It seems weird to me that we're coupling the payload and a max size here with anything HTTP header related, unless there's some sort of limit on server??